packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js

```diff
@@ -777,14 +810,96 @@ describe('ReactComponentLifeCycle', () => {
     }
 
     expect(() => ReactDOM.render(<WillReceiveProps />, container)).toWarnDev(
-      'Unsafe legacy lifecycles will not be called for components using the new getDerivedStateFromProps() API.\n\n' +
+      'Unsafe legacy lifecycles will not be called for components using new component APIs.\n\n' +
         'WillReceiveProps uses getDerivedStateFromProps() but also contains the following legacy lifecycles:\n' +
         '  componentWillReceiveProps\n\n' +
         'The above lifecycles should be removed. Learn more about this warning here:\n' +
         'https://fb.me/react-async-component-lifecycle-hooks',
     );
   });
 
+  it('should warn about deprecated lifecycles (cWM/cWRP/cWU) if new getSnapshotBeforeUpdate is present', () => {
+    const container = document.createElement('div');
+
+    class AllLegacyLifecycles extends React.Component {
+      state = {};
+      getSnapshotBeforeUpdate() {}
+      componentWillMount() {}
+      UNSAFE_componentWillReceiveProps() {}
```

Comment:
Why use `UNSAFE_componentWillReceiveProps` in this test? Just curious—doesn't seem like the test is specifically about `UNSAFE_componentWillReceiveProps`.

Reply:
I just tried to cover a mix of old and new aliases in the test. There's not really a meaningful pattern to it.

Reply:
To be clear, these tests are kind of verbose and I was hoping to avoid having to duplicate them for old and new lifecycle aliases, so I just tried to cover a mix of both (so the test could verify that the correct lifecycle name was printed in the warning)